rfcs/0000-limitador-multithread-inmemory.md

```diff
@@ -0,0 +1,241 @@
+# RFC Template
+
+- Feature Name: `limitador_multithread_inmemory`
+- Start Date: 2023-11-02
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+
+# Summary
+[summary]: #summary
+
+Enable Limitador service to process requests in parallel when configured to use In Memory storage.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Limitador service is single threaded, regardless of its chosen storage. This means that it can only process one request at a time. 
+Alternatively, we could use a multithreading approach to process requests in parallel, which would improve the overall performance of Limitador.
+However, this would introduce some particular behaviour regarding the _Accuracy_ of the defined limit counters and the _Throughput_ of the service.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+In order to achieve the desired multi-threading functionality, we'd need to make sure that the user understands the
+trade-offs that this approach implies and that they are aware of the possible consequences of using it, and how to
+configure the service in order to obtain the desired behaviour. Regarding this last statement, we will need to
+introduce an interface that allows the user to meet their requirements.
+
+A couple of concepts we need to define in order to understand the proposal:
+* **Accuracy**: The _accuracy_ is defined by the service's ability to enforce the limits in an exact factual way. This means
+  that if the service is configured to allow 10 requests per second, it will only allow 10 requests per second, and
+  not 11 or 9. In terms of observability, the counters, will be monotonically strictly decreasing without repeating any
+  value given per service.
+* **Throughput**: The _throughput_ of a service is defined by the number of requests that the service can process in a
+  given time. As a consequence of having higher throughput, we need to introduce two more concepts:
+  * **Overshoot**: The _overshooting_ of a limit counter is defined by the difference between the _expected_ value of the counter
```

Comment:
_overshooting_ is when the counter goes beyond the limit and the request is still admitted, right?

Reply:
Interesting, I agree with your definition, but looks like @didierofrivia looks at it the other way around.